server/channels/app/channel.go

```diff
@@ -1680,6 +1709,57 @@ func (a *App) AddUserToChannel(c request.CTX, user *model.User, channel *model.C
 	return newMember, nil
 }
 
+func (a *App) addChannelToDefaultCategory(c request.CTX, user *model.User, channel *model.Channel) {
+	// Add channel to default category if specified
+	if channel.DefaultCategoryName != "" && *a.Config().ExperimentalSettings.ExperimentalChannelCategorySorting {
+		mlog.Info("Adding channel to default category", mlog.String("user_id", user.Id), mlog.String("team_id", channel.TeamId), mlog.String("channel_id", channel.Id), mlog.String("category_name", channel.DefaultCategoryName))
+		// Get user's categories for this team
+		categories, err := a.GetSidebarCategoriesForTeamForUser(c, user.Id, channel.TeamId)
+		if err != nil {
+			mlog.Error("Failed to get sidebar categories", mlog.String("user_id", user.Id), mlog.String("team_id", channel.TeamId), mlog.Err(err))
+			c.Logger().Warn("Failed to get sidebar categories", mlog.String("user_id", user.Id), mlog.String("team_id", channel.TeamId), mlog.Err(err))
```

Comment:
Mistake? 

Also, let's log and return. It will let us outdent the else block.

Reply:
I think some of these extra logs were added by AI and I didn't notice, that's on me.
I've removed these and the other extraneous ones.